test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -9932,10 +9932,10 @@ BOOST_AUTO_TEST_CASE(packed_functions)
 	char const* sourceCode = R"(
 		contract C {
 			// these should take the same slot
-			function() external returns (uint) a;
```

Comment:
Why is this change needed?

Reply:
Compiling the contract would fail because ``g``'s type is not implicitly convertible to ``function() external returns (uint)``. This can be fixed by using ``d  = this.g`` and ``a = this.g``, but I still do not fully understand why. Could you quickly elaborate on this?

Reply:
Ok, now I understand why adding ``this`` works.